### ISSUE TITLE
`readonly` and `declare` invalid property for `MethodDefinition`.

### ISSUE THREAD

**Issue by sosukesuzuki** [typescript-eslint/typescript-eslint#2908]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```TS
// the code you're trying to parse
class Foo {
  readonly bar() {}
  declare baz() {}
}
```

```TS
// the code you're using to do the parse of the aforementioned code
import { parse } from '@typescript-eslint/typescript-estree';

const code = `
class Foo {
  readonly bar() {}
  declare baz() {}
}
`;
const ast = parse(code);
```

**Expected Result**

Expect that `MethodDefinition` nodes have `readonly` and `declare` property.

<details>
  <summary>AST</summary>
  <div>

```diff
{
  "type": "ClassDeclaration",
  "id": {
    "type": "Identifier",
    "name": "Foo"
  },
  "body": {
    "type": "ClassBody",
    "body": [
      {
        "type": "MethodDefinition",
        "key": {
          "type": "Identifier",
          "name": "bar"
        },
        "value": {
          "type": "FunctionExpression",
          "id": null,
          "generator": false,
          "expression": false,
          "async": false,
          "body": {
            "type": "BlockStatement",
            "body": []
          },
          "params": []
        },
        "computed": false,
        "static": false,
        "kind": "method",
+        "readonly": true
      },
      {
        "type": "MethodDefinition",
        "key": {
          "type": "Identifier",
          "name": "baz"
        },
        "value": {
          "type": "FunctionExpression",
          "id": null,
          "generator": false,
          "expression": false,
          "async": false,
          "body": {
            "type": "BlockStatement",
            "body": []
          },
          "params": []
        },
        "computed": false,
        "static": false,
        "kind": "method",
+       "declare": true
      }
    ]
  },
  "superClass": null
}
```

  </div>
</details>

**Actual Result**

<details>
  <summary>AST</summary>
  <div>

```json
{
  "type": "ClassDeclaration",
  "id": {
    "type": "Identifier",
    "name": "Foo"
  },
  "body": {
    "type": "ClassBody",
    "body": [
      {
        "type": "MethodDefinition",
        "key": {
          "type": "Identifier",
          "name": "bar"
        },
        "value": {
          "type": "FunctionExpression",
          "id": null,
          "generator": false,
          "expression": false,
          "async": false,
          "body": {
            "type": "BlockStatement",
            "body": []
          },
          "params": []
        },
        "computed": false,
        "static": false,
        "kind": "method"
      },
      {
        "type": "MethodDefinition",
        "key": {
          "type": "Identifier",
          "name": "baz"
        },
        "value": {
          "type": "FunctionExpression",
          "id": null,
          "generator": false,
          "expression": false,
          "async": false,
          "body": {
            "type": "BlockStatement",
            "body": []
          },
          "params": []
        },
        "computed": false,
        "static": false,
        "kind": "method"
      }
    ]
  },
  "superClass": null
}
```

</details>

**Additional Info**

https://github.com/prettier/prettier/issues/9760

I know the above code is invalid as TypeScript program ([TS Playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8CwAoa0BOBTMAJvAHYgCe0ARmNgBQCUKAvhlgbqDblWAF4PMMLdEA)). But I think code formatters should correctly treat those cases. What do you think about?

If the request is accepted, I can work on this.

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `master` |
| `TypeScript`                           | `4.1.2` |
| `node`                                 | `v14.8.0` |

**Comment by bradzacher** [comment]
Whilst the parser treats them as valid (because we throw no errors currently https://github.com/typescript-eslint/typescript-eslint/issues/1852) they shouldn't be included in the AST.

I think this falls into the same category as decorators on things that aren't classes - we don't error for them, we don't emit AST for them, and consumers can use the underlying TS AST to look for them if they want.

With the next major we'll start properly handling syntax errors, so this won't be an issue.

**Comment by sosukesuzuki** [comment]
I got it. Certainly, it seems unreasonable for typescript-estree to address this problem.
